src/Elliptic/Systems/SelfForce/GeneralRelativity/Equations.hpp

```diff
@@ -0,0 +1,130 @@
+            const GradTensorType& field_gradient);
+
+/*!
+ * \brief The first-order flux on an element face
+ * $F^i=\{n_{r_\star}, \alpha n_\theta\}\Psi_m$.
+ */
+void fluxes_on_face(gsl::not_null<FluxTensorType*> flux,
+                    const Scalar<ComplexDataVector>& alpha,
+                    const tnsr::I<DataVector, 2>& face_normal_vector,
+                    const tnsr::aa<ComplexDataVector, 3>& field);
+
+/*!
+ * \brief The source term $\beta_{ab}^{cd} (\Psi_m)_{cd} + \gamma_{iab}^{cd}
+ * F^i_{cd}$.
+ */
+void add_sources(gsl::not_null<tnsr::aa<ComplexDataVector, 3>*> source,
+                 const tnsr::aaBB<ComplexDataVector, 3>& beta,
+                 const tnsr::aaBB<ComplexDataVector, 3>& gamma_rstar,
+                 const tnsr::aaBB<ComplexDataVector, 3>& gamma_theta,
+                 const tnsr::aa<ComplexDataVector, 3>& field,
+                 const FluxTensorType& flux);
+
+/// Fluxes $F^i$ for the gravitational self-force system.
+/// \see GrSelfForce::FirstOrderSystem
+struct Fluxes {
+  using argument_tags = tmpl::list<Tags::Alpha>;
+  using volume_tags = tmpl::list<>;
+  using const_global_cache_tags = tmpl::list<>;
+  static constexpr bool is_trivial = false;
+  static constexpr bool is_discontinuous = false;
+  static void apply(gsl::not_null<FluxTensorType*> flux,
+                    const Scalar<ComplexDataVector>& alpha,
+                    const tnsr::aa<ComplexDataVector, 3>& /*field*/,
+                    const GradTensorType& field_gradient);
+  static void apply(gsl::not_null<FluxTensorType*> flux,
+                    const Scalar<ComplexDataVector>& alpha,
+                    const tnsr::i<DataVector, 2>& /*face_normal*/,
+                    const tnsr::I<DataVector, 2>& face_normal_vector,
+                    const tnsr::aa<ComplexDataVector, 3>& field);
+};
+
+/// Source terms for the gravitational self-force system.
+/// \see GrSelfForce::FirstOrderSystem
+struct Sources {
+  using argument_tags =
+      tmpl::list<Tags::Beta, Tags::GammaRstar, Tags::GammaTheta>;
+  using const_global_cache_tags = tmpl::list<>;
+  static void apply(
+      gsl::not_null<tnsr::aa<ComplexDataVector, 3>*> scalar_equation,
+      const tnsr::aaBB<ComplexDataVector, 3>& beta,
+      const tnsr::aaBB<ComplexDataVector, 3>& gamma_rstar,
+      const tnsr::aaBB<ComplexDataVector, 3>& gamma_theta,
+      const tnsr::aa<ComplexDataVector, 3>& field, const FluxTensorType& flux);
+};
+
+/*!
+ * \brief Adds or subtracts the singular field to/from the received data on
+ * element boundaries.
+ *
+ * In the regularized region we solve for the regularized field
+ * \begin{equation}
+ *   \Psi_m^R = \Psi_m - \Psi_m^P
+ *   \text{,}
+ * \end{equation}
+ * so we subtract the singular field on the regularized side and add it on the
+ * other side of the boundary. We do the same for the received normal dot flux
+ * $n_i F^i$, but with an extra minus sign because this quantity is defined with
```

Comment:
Could you document what the argument `field_is_regularized` corresponds to? E.g. if `field_is_regularized` is `true` does that mean you add the singular term or subtract it?

Reply:
added and squashed in